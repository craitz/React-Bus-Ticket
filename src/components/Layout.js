import React, { Component } from 'react';
import { Grid, Row, Col } from 'react-bootstrap';
import Navigation from '../shared/Navigation';

class Layout extends Component {
  render() {
    return (
      <div className="App">
        <Navigation></Navigation>
        {this.props.children}
      </div>
    );
  }
}

export default Layout;