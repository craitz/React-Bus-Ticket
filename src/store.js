import { applyMiddleware, createStore } from 'redux';
import { createLogger } from 'redux-logger';
import thunk from 'redux-thunk';
import promise from 'redux-promise-middleware';
import reducer from './reducers';

const configureStore = () => {
  const logger = createLogger({
    collapsed: true,
    duration: true,
    diff: true,
    level: ''
  });

  const middleware = applyMiddleware(promise(), thunk, logger);

  return createStore(reducer, middleware);
}

export default configureStore();
