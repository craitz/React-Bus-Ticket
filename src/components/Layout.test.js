import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom'
import { Provider } from 'react-redux';
import store from '../store';
import renderer from 'react-test-renderer';
// import Navigation from '../shared/Navigation';
import Layout from './Layout';

describe('Layout VIEW', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <Router>
        <Provider store={store}>
          <Layout />
        </Provider>
      </Router>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});