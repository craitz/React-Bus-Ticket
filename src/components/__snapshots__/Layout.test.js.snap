// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout VIEW renders correctly 1`] = `
<div
  className="App"
>
  <nav
    className="navbar navbar-inverse"
  >
    <div
      className="container"
    >
      <div
        className="navbar-header"
      >
        <span
          className="navbar-brand"
        >
          <span
            aria-hidden={true}
            className="fa fa-bus App-logo"
          />
          <span
            className="header-title"
          >
            BusTicket
          </span>
        </span>
        <button
          className="navbar-toggle collapsed"
          onClick={[Function]}
          type="button"
        >
          <span
            className="sr-only"
          >
            Toggle navigation
          </span>
          <span
            className="icon-bar"
          />
          <span
            className="icon-bar"
          />
          <span
            className="icon-bar"
          />
        </button>
      </div>
    </div>
  </nav>
</div>
`;
